index.js

```diff
@@ -191,36 +191,42 @@ jasmine.Expectation.prototype.wrapCompare = function(name, matcherFactory) {
 
       var result = matcherCompare.apply(null, args);
 
-      return webdriver.promise.when(result.pass).then(function(pass) {
-        var message = '';
+      if (webdriver.promise.isPromise(result.pass)) {
```

Comment:
Why this change?  Even if `result.pass` weren't a promise `webdriver.promise.when` should still work shouldn't it?